### DISCUSSION TITLE
Why is LocalDateTime used rather than Instant for FIX UTCTimestamps?

### DISCUSSION THREAD

**Discussion by abissell** [quickfix-j/quickfixj#629] [Q&A]
I tried figuring this out for myself from documentation, code, and discussion on Github but couldn't seem to find an answer. I was wondering, why is `LocalDateTime` used for FIX UTCTimestamp fields like `SendingTime`? From my cursory reading about the two different types, `LocalDateTime` is not intended to carry any timezone information with it, while `Instant` represents a moment on the UTC timeline, which seems to correspond more closely to the FIX spec definition of these fields.

Appreciate any insights you can offer, thanks!

**Comment by abissell** [comment]
BTW I hope this doesn't come across as language lawyering or anything, the reason I ask is I'm writing a library to serve as an intermediate layer between Quickfix/J and other Java FIX implementations, and just want to pick the best data type for the UTC timestamp fields. Thanks

**Reply by chrjohn** [reply]
No problem, it is a good question indeed. :)
I shortly thought about it and couldn't find a good reason. I'll give it another thought after the weekend. But it'll probably turn out as ignorance.

**Reply by chrjohn** [reply]
This was some years ago, maybe I thought `LocalTime` with `UtcTimeOnly` fields vs `LocalDateTime` with `UtcTimestamp` field was a good idea.
However, please feel free to open a corresponding issue or PR. Thank you. :)
